flink-learning-common/src/test/java/com/zhisheng/common/utils/DateUtilTests.java

```diff
@@ -0,0 +1,60 @@
+package com.zhisheng.common.utils;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+import java.util.Date;
+
+import static com.zhisheng.common.utils.DateUtil.*;
+import static org.junit.Assert.*;
+
+public class DateUtilTests {
+
+    @Test
+    public void testFormat() {
+        Date date1 = new Date(1556665200000L);
+        Date date2 = new Date(1556723220000L);
+        Date date3 = new Date(1556723235000L);
+        Date date4 = new Date(1556723235000L);
+
+        assertEquals(date1.getTime(), DateUtil.format(date1));
+
+        assertEquals("2019-05-01", DateUtil.format(date1.getTime(), YYYY_MM_DD));
+        assertEquals("2019-05-01", DateUtil.format(date1, DateTimeZone.getDefault(), YYYY_MM_DD));
+
+        assertEquals(date1.getTime(), DateUtil.format("2019-05-01", YYYY_MM_DD));
+        assertEquals(date2.getTime(), DateUtil.format("2019-05-01 16:07", YYYY_MM_DD_HH_MM));
+        assertEquals(date3.getTime(), DateUtil.format("2019-05-01 16:07:15", YYYY_MM_DD_HH_MM_SS));
+        assertEquals(date4.getTime(), DateUtil.format("2019-05-01 16:07:15.0", YYYY_MM_DD_HH_MM_SS_0));
+    }
+
+    @Test
+    public void testIsValidDate() {
+        assertTrue(isValidDate("2019-05-01", YYYY_MM_DD));
+
+        assertFalse(isValidDate("01-05-2019", YYYY_MM_DD));
+    }
+
+    @Test
+    public void testToDate() {
+        assertEquals(new Date(1556665200000L), toDate("2019-05-01", YYYY_MM_DD));
+    }
+
+    @Test
+    public void testWithTimeAtStartOfDay() {
+        assertEquals("2019-05-01", withTimeAtStartOfDay(new Date(), YYYY_MM_DD));
+        assertEquals("2019-05-01", withTimeAtStartOfDay(new DateTime(), YYYY_MM_DD));
+    }
+
+    @Test
+    public void testwithTimeAtEndOfDay() {
+        assertEquals("2019-05-01", withTimeAtEndOfDay(new Date(), YYYY_MM_DD));
+        assertEquals("2019-05-01", withTimeAtEndOfDay(new DateTime(), YYYY_MM_DD));
+    }
+
+    @Test
+    public void testwithTimeAtStartOfNow() {
+        assertEquals("20190501000000", withTimeAtStartOfNow());
+        assertEquals("20190501235959", withTimeAtEndOfNow());
```

Comment:
Wouldn't these values change?

Reply:
Anything with date and time is mocked by default in Diffblue Cover. Did you refactor these changes out?

Reply:
Yup they change daily `facepalm`.. I added the mocks from Diffblue cover. I haven't used mocks before and I am not super confident with it... so there maybe a few things I have missed. Also the code tested by each mock is only a single line method and because of that wouldn't `Assert.assertNotNull` be more cleaner?

I also know that the lines are longer than 80 characters. Will change that later :)

Reply:
Preset dates created